results/cmd/api/main_test.go

```diff
@@ -0,0 +1,37 @@
+package main
+
+import (
+	"context"
+	"log"
+	"testing"
+	"time"
+
+	pb "github.com/tektoncd/experimental/results/proto/proto"
+	"google.golang.org/grpc"
+)
+
+const (
+	address = "localhost:50051"
+)
+
+// Test functionality of Server code
+func TestCreateTaskRun(t *testing.T) {
+	// Set up a connection to the server.
```

Comment:
instead of connecting to a server over its port and then running the test, it should be possible to just test the CreateTaskRun method:

```
  tmpfile, err := ioutil.TempFile("", "testdb")
  if err != nil {
		t.Fatalf("failed to create temp file for db: %v", err)
  }

  defer os.Remove(tmpfile.Name()) // clean up at the end of test
  testDB, err := setupTestDB(tmpfile.Name()) // create a helper function that opens the db and adds the schema like in main.go
  if err != nil { ... } //handle error
  
  srv := srv := &server{db: testDb}

  // And then our test as before
   ...
   r, err := srv.CreateTaskRun(ctx, .....)
```

Reply:
resolved this issue